src/org/omegat/core/dictionaries/LingvoDSL.java

```diff
@@ -130,4 +134,91 @@ private void loadData(Stream<String> stream) {
                     .collect(Collectors.toList());
         }
     }
+
+    @SuppressWarnings("visibilitymodifier")
+    static class RE {
+        public Pattern pattern;
+        public String replacement;
+
+        public RE(final String regex, final String replacement) {
+            pattern = Pattern.compile(regex);
+            this.replacement = replacement;
+        }
+    }
+
+    static class LingvoDSLTag {
+        private static final List<RE> RE_LIST;
+
+        static String replaceTag(final String line) {
+            String result = line;
+            for (RE re : RE_LIST) {
+                result = re.pattern.matcher(result).replaceAll(re.replacement);
+            }
+            return result;
+        }
+
+        static {
+            List<RE> reList = new ArrayList<>();
+            reList.add(new RE("\\[\\[(.+?)\\]\\]", "&#91;$1&#93;"));
+            reList.add(new RE("\\\\\\[", "&#91;"));
+            reList.add(new RE("\\\\\\]", "&#93;"));
+            reList.add(new RE("\\[b\\](.+?)\\[/b\\]", "<strong>$1</strong>"));
+            reList.add(new RE("\\[i\\](.+?)\\[/i\\]", "<span style='font-style: italic'>$1</span>"));
+            reList.add(new RE("\\[trn\\](.+?)\\[/trn\\]", "$1"));
+            reList.add(new RE("\\[t\\](.+?)\\[/t\\]", "$1&nbsp;"));
+            reList.add(new RE("\\[br\\]", "<br/>"));
+            // Green is default color in Lingvo
+            reList.add(new RE("\\[c\\](.+?)\\[/c\\]", "<span style='color:green'>$1</span>"));
+            // The following line tries to replace [c value]text[/c] with text colored as per the value.
+            // Since the color names are plain words like 'red', or 'blue', or 'steelgray' etc.,
+            // FIXME: I use the ([a-z]+?) regular expression, but am not sure if it is correct.
+            reList.add(new RE("\\[c\\s([a-z]+?)\\](.+?)\\[/c\\]", "<span style='color:$1'>$2</span>"));
+            reList.add(new RE("\\[com\\]", ""));
+            reList.add(new RE("\\[/com\\]", ""));
+            reList.add(new RE("\\[ex\\]", ""));
+            reList.add(new RE("\\[/ex\\]", ""));
+            reList.add(new RE("\\[lang\\]", ""));
+            reList.add(new RE("\\[/lang\\]", ""));
```

Comment:
You can combine these into a single regex.

Reply:
Actually for consistency wouldn't it make sense to do these like the other ones?

```java
reList.add(new RE("\\[lang\\](.+?)\\[/lang\\]", "$1"));
```

Reply:
- In fact you could combine all of the "simple replacement" ones like below
- I highly recommend using [named capturing groups](https://docs.oracle.com/javase/8/docs/api/java/util/regex/Pattern.html#groupname)

```java
reList.add(new RE("\\[(?<tag>com|ex|lang|...)\\](?<content>.+?)\\[/\ktag\\]", "${content}"));
```